endpoint.cpp

```diff
@@ -347,7 +347,42 @@ void Endpoint::log_aggregate(unsigned int interval_sec)
     }
 }
 
-int UartEndpoint::open(const char *path, speed_t baudrate)
+int UartEndpoint::set_speed(speed_t baudrate)
+{
+    struct termios2 tc;
+
+    if (fd < 0) {
+        return -1;
+    }
+
+    bzero(&tc, sizeof(tc));
+    if (ioctl(fd, TCGETS2, &tc) == -1) {
+        log_error_errno(errno, "Could not get termios2 (%m)");
+        return -1;
+    }
+
+    /* speed is configured by c_[io]speed */
+    tc.c_cflag &= ~CBAUD;
+    tc.c_cflag |= BOTHER;
+    tc.c_ispeed = baudrate;
+    tc.c_ospeed = baudrate;
+
+    if (ioctl(fd, TCSETS2, &tc) == -1) {
+        log_error_errno(errno, "Could not set terminal attributes (%m)");
+        return -1;
+    }
+
+    if (ioctl(fd, TCFLSH, TCIOFLUSH) == -1) {
+        log_error_errno(errno, "Could not flush terminal (%m)");
+        return -1;
+    }
+
+    log_info("UART [%d] speed = %u", fd, baudrate);
```

Comment:
If the flush doesn't happen, terminal speed may still change, right? If so, we may want to log_info to which speed we are changing anyway - iff things work even if flush fails.
If last sentence is not true, ignore this comment.